python/tests/backends/test_braket.py

```diff
@@ -0,0 +1,292 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+import cudaq
+import pytest
+import os
+from cudaq import spin
+import numpy as np
+
+## NOTE: Comment the following line which skips these tests in order to run in
+# local dev environment after setting AWS credentials.
+## NOTE: Amazon Braket costs apply
+pytestmark = pytest.mark.skip("Amazon Braket credentials required")
+
+
+@pytest.fixture(scope="session", autouse=True)
+def do_something():
+    device_arn = "arn:aws:braket:::device/quantum-simulator/amazon/sv1"
+    cudaq.set_target("braket", machine=device_arn)
+    yield "Running the tests."
+    cudaq.__clearKernelRegistries()
+    cudaq.reset_target()
+
+
+def test_simple_kernel():
+
+    @cudaq.kernel
+    def kernel():
+        q = cudaq.qubit()
+        x(q)
+        mz(q)
+
+    counts = cudaq.sample(kernel, shots_count=100)
+    assert len(counts) == 1
+    assert "1" in counts
+
+
+def test_multi_qubit_kernel():
+
+    @cudaq.kernel
+    def kernel():
+        q0 = cudaq.qubit()
+        q1 = cudaq.qubit()
+        h(q0)
+        cx(q0, q1)
+        mz(q0)
+        mz(q1)
+
+    with pytest.raises(RuntimeError) as e:
+        cudaq.sample(kernel, shots_count=100)
+    assert "cannot declare a qubit register. Only 1 qubit register(s) is/are supported" in repr(
+        e)
+
+
+def test_qvector_kernel():
+
+    @cudaq.kernel
+    def kernel():
+        qubits = cudaq.qvector(2)
+        h(qubits[0])
+        cx(qubits[0], qubits[1])
+        mz(qubits)
+
+    counts = cudaq.sample(kernel, shots_count=100)
+    assert len(counts) == 2
+    assert "00" in counts
+    assert "11" in counts
+
+
+def test_builder_sample():
+
+    kernel = cudaq.make_kernel()
+    qubits = kernel.qalloc(2)
+    kernel.h(qubits[0])
+    kernel.cx(qubits[0], qubits[1])
+    kernel.mz(qubits)
+    counts = cudaq.sample(kernel, shots_count=100)
+    assert len(counts) == 2
+    assert "00" in counts
+    assert "11" in counts
+
+
+def test_all_gates():
+
+    @cudaq.kernel
+    def single_qubit_gates():
```

Comment:
It would be good to also add a test for all the singly-controlled gates, and another test for all the adjoints.

Reply:
Done. Updated `test_control_modifier` and `test_adjoint_modifier`